src/OrbitGl/TimeGraph.cpp

```diff
@@ -74,18 +74,43 @@ TimeGraph::TimeGraph(AccessibleInterfaceProvider* parent, OrbitApp* app,
 
   const orbit_client_data::ModuleManager* module_manager =
       app != nullptr ? app->GetModuleManager() : nullptr;
+
   track_container_ = std::make_unique<orbit_gl::TrackContainer>(this, this, viewport, &layout_,
                                                                 app_, module_manager, capture_data);
   timeline_ui_ = std::make_unique<orbit_gl::TimelineUi>(
       /*parent=*/this, /*timeline_info_interface=*/this, viewport, &layout_);
+
+  slider_ = std::make_shared<orbit_gl::GlHorizontalSlider>(
+      /*parent=*/this, /*timeline_info_interface*/ this, viewport, &layout_);
+  vertical_slider_ = std::make_shared<orbit_gl::GlVerticalSlider>(
+      /*parent=*/this, /*timeline_info_interface*/ this, viewport, &layout_);
+
+  slider_->SetDragCallback([&](float ratio) {
+    this->UpdateHorizontalScroll(ratio);
+    RequestUpdate();
+  });
+  slider_->SetResizeCallback([&](float normalized_start, float normalized_end) {
+    this->UpdateHorizontalZoom(normalized_start, normalized_end);
+    RequestUpdate();
+  });
+
+  vertical_slider_->SetDragCallback([&](float ratio) {
+    track_container_->UpdateVerticalScroll(ratio);
+    RequestUpdate();
+  });
+
+  vertical_slider_->SetOrthogonalSliderPixelHeight(slider_->GetPixelHeight());
+  slider_->SetOrthogonalSliderPixelHeight(vertical_slider_->GetPixelHeight());
+
+  // Needed because unit-tests rely on children positions before drawing.
+  SetWidth(viewport_->GetWorldWidth());
+  UpdateChildrenPosAndContainerSize();
```

Comment:
You should be able to get around this with the new utility methods in `CaptureViewElementTester` (not yet merged). I guess it's fine to keep it, but there should be a TODO.

Reply:
It's done! Thanks for PreRender()! :)

Reply:
Do you still need those two lines then?

Reply:
No, I don't really need them. I was sure I had erased them. Thanks for catching!